Create new `header` package

dagoof opened the pull request:
## Describe your changes

Include screenshots, photos or links if necessary.

## Checklist

- [ ] Read and check your code before tagging someone for review.
- [x] Run `yarn format`
- [x] Run `yarn lint` in the root of the repository to ensure tests are passing
- [x] Add necessary tests
- [x] Run `yarn changeset` to create a changeset file
- [x] Write documentation (README.md)
- [x] Create stories for Storybook

dagoof committed:
rebase me

dagoof committed:
rebase me

dagoof committed:
active path

dagoof committed:
rebase me

dagoof committed:
rebase me

jordanoverbye commented on packages/header/src/Header.tsx:
```diff
@@ -0,0 +1,71 @@
+			activePath={activePath}
+			rightContent={
+				authenticated ? (
+					<MainNavLink
```
Since this has an `onClick` and no `href`, we should use a button instead of a link.

```suggestion
					<MainNavButton
```

jordanoverbye commented on packages/header/src/Header.tsx:
```diff
@@ -0,0 +1,71 @@
+						icon={AvatarIcon}
+					/>
+				) : (
+					<MainNavLink
```
```suggestion
					<MainNavButton
```

jordanoverbye committed:
Storybook stories

jordanoverbye committed:
props

jordanoverbye committed:
change href

dagoof committed:
rebase me

dagoof commented on packages/header/src/Header.tsx:
```diff
@@ -0,0 +1,71 @@
+export type HeaderProps = {
+	authenticated: boolean;
+	activePath: string;
+	handleSignIn?: React.MouseEventHandler<HTMLButtonElement>;
```
what do you think about this -- Is the consumer of this component concerned with the triggering element?

the original promise<void> signaled to the user the intent of how the property should be used. 

interested to hear your thoughts

dagoof committed:
clean me

dagoof committed:
rewind loading changes

dagoof committed:
rebase me

dagoof committed:
yarnlock

dagoof committed:
Header: add header package & documentation

Header exposes a component that implements the default export service header
made up of a design system components

It includes links to the exort services applications

jordanoverbye committed:
Changeset

jordanoverbye committed:
Remove basic stories

jordanoverbye committed:
Update auth state

dagoof committed:
add loading dep to header
